File: get-set-ticket/get-set-ticket/get-set-ticket-frontEnd/get-set-ticket/src/app/model/Ticket.ts
export class Ticket {
    constructor(
        public ticketId:string,
        public userId:string,
        public source:string,
        public destination:string,
        public journeyStartDate:Date,
        public journeyEndDate:Date,
        public name:string,
        public age:number,
        public gender:string,
        public pnrNo:string,
        public trainNo:string,
        public trainName:string,
        public action:string
    ) {}
}

File: get-set-ticket/get-set-ticket/get-set-ticket-backEnd/get-set-ticket/src/main/java/myapp/getsetticket/entity/Ticket.java
package myapp.getsetticket.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.*;
import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;

@Entity(name = "ticket_table")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Ticket {
    @Id
    private String ticketId;
    private String userId;
    private String source;
    private String destination;
    private @DateTimeFormat(pattern = "yyyy-MM-dd") Date journeyStartDate;
    private @DateTimeFormat(pattern = "yyyy-MM-dd") Date journeyEndDate;
    private String name;
    private Long age;
    private String gender;
    private String pnrNo;
    private String trainNo;
    private String trainName;
    private String action;
}
